src/wp-includes/abilities-api.php

```diff
@@ -0,0 +1,192 @@
+ * @subpackage Abilities_API
+ * @since 6.9.0
+ */
+
+declare( strict_types = 1 );
+
+/**
+ * Registers a new ability using Abilities API.
+ *
+ * Note: Should only be used on the {@see 'abilities_api_init'} hook.
+ *
+ * @since 6.9.0
+ *
+ * @see WP_Abilities_Registry::register()
+ *
+ * @param string              $name The name of the ability. The name must be a string containing a namespace
+ *                                  prefix, i.e. `my-plugin/my-ability`. It can only contain lowercase
+ *                                  alphanumeric characters, dashes and the forward slash.
+ * @param array<string,mixed> $args {
+ *     An associative array of arguments for the ability.
+ *
+ *     @type string               $label                 The human-readable label for the ability.
+ *     @type string               $description           A detailed description of what the ability does.
+ *     @type string               $category              The category slug this ability belongs to.
+ *     @type callable             $execute_callback      A callback function to execute when the ability is invoked.
+ *                                                       Receives optional mixed input and returns mixed result or WP_Error.
+ *     @type callable             $permission_callback   A callback function to check permissions before execution.
+ *                                                       Receives optional mixed input and returns bool or WP_Error.
+ *     @type array<string,mixed>  $input_schema          Optional. JSON Schema definition for the ability's input.
+ *     @type array<string,mixed>  $output_schema         Optional. JSON Schema definition for the ability's output.
+ *     @type array<string,mixed>  $meta                  {
+ *         Optional. Additional metadata for the ability.
+ *
+ *         @type array<string,bool|string> $annotations  Optional. Annotation metadata for the ability.
+ *         @type bool                      $show_in_rest Optional. Whether to expose this ability in the REST API. Default false.
+ *     }
+ *     @type string               $ability_class         Optional. Custom class to instantiate instead of WP_Ability.
+ * }
+ * @return WP_Ability|null An instance of registered ability on success, null on failure.
+ *
+ * @phpstan-param array{
+ *   label?: string,
+ *   description?: string,
+ *   category?: string,
+ *   execute_callback?: callable( mixed $input= ): (mixed|\WP_Error),
+ *   permission_callback?: callable( mixed $input= ): (bool|\WP_Error),
+ *   input_schema?: array<string,mixed>,
+ *   output_schema?: array<string,mixed>,
+ *   meta?: array{
+ *     annotations?: array<string,(bool|string)>,
+ *     show_in_rest?: bool,
+ *     ...<string,mixed>,
+ *   },
+ *   ability_class?: class-string<\WP_Ability>,
+ *   ...<string, mixed>
+ * } $args
+ */
+function wp_register_ability( string $name, array $args ): ?WP_Ability {
+	if ( ! did_action( 'abilities_api_init' ) ) {
```

Comment:
Maybe sort of a nit-pick: While WordPress Core hooks are certainly inconsistent in whether they include the `wp_` prefix or not, in recent years it's been strongly encouraged to always use the `wp_` prefix. So I think we should apply that to all actions and filters in this PR.